packages/pwa-kit-react-sdk/src/ssr/browser/main.jsx

```diff
@@ -132,10 +132,40 @@ export const start = async () => {
         locals
     })
 
+    // Await the async getRoutes function
+    let routes = await getRoutes(locals)
+
+    let serializedRoutes = window.__CONFIG__.app.routes
+    // Deserialize routes
+    serializedRoutes = serializedRoutes.map(
+        ({path, componentName, componentProps}) => {
+            let component = routes.find((route) =>
+                route.component?.displayName?.includes(componentName)
+            )?.component
+            if (!component) {
+                // TODO: Error handling if given component couldn't be found
+                console.error('Component', componentName, 'could not be deserialized for path', path)
+                return
+            }
+
+            if (componentProps) {
+                const Component = component
+                component = () => <Component {...componentProps} />
+            }
+            return {
+                path,
+                exact: true,
+                component
+            }
+        }
+    )
+    serializedRoutes = serializedRoutes.filter((route) => !!route)
+    routes = serializedRoutes
```

Comment:
@bendvc Moved the deserialization to here so it only runs on client-side

Reply:
Interesting.. would the async logic for making your seo url mapping called be called now on the client even though you have the routes serialized. 

E.g. on line 135 you get the routes which will call the extendRoutes for all extensions and await on it.. then you will deserialize the routes starting at line 138. Effectively coming up with the same route list?

Reply:
Good point it does seem unnecessary to deserialize routes that are already in the routes array. 

I'll look to only serialize/deserialize the route of the current path